### PR TITLE
Update to xcproj 4.0

### LINKED ISSUE #223: Update to xcproj 2.0 or later
Are there plans to upgrade to xcproj 2.0? Are there any known issues/blockers? I want to add XcodeGen support for scheme pre- and post- actions, which will depend on xcodeswift/xcproj#217, which will presumably be in a 2.x.y release.

### PR DESCRIPTION
**Pull request by yonaskolb**
Resolves #223 

This mostly involves dealing with the fact that PBXObject no longer has a `reference`

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ let package = Package(
         .package(url: "https://github.com/yonaskolb/JSONUtilities.git", from: "3.3.0"),
         .package(url: "https://github.com/kylef/Spectre.git", from: "0.8.0"),
         .package(url: "https://github.com/onevcat/Rainbow.git", from: "3.0.0"),
-        .package(url: "https://github.com/xcodeswift/xcproj.git", from: "1.7.0")
+        .package(url: "https://github.com/xcodeswift/xcproj.git", .revision("ee552d3"))
```
Comment (rahul-malik) [inline Package.swift]: Pin to version?

File: Sources/ProjectSpec/Target.swift [hunk 2]
```diff
@@ -219,7 +218,7 @@ extension Target: NamedJSONDictionaryConvertible {
 
     public init(name: String, jsonDictionary: JSONDictionary) throws {
         self.name = jsonDictionary.json(atKeyPath: "name") ?? name
-        self.productName = jsonDictionary.json(atKeyPath: "productName")
+        productName = jsonDictionary.json(atKeyPath: "productName")
```
Comment (rahul-malik) [inline Sources/ProjectSpec/Target.swift]: just removing `self.`? should we do this on the line above as well (`self.name`) ?

File: Sources/XcodeGenKit/ProjectGenerator.swift [hunk 3]
```diff
@@ -31,24 +31,21 @@ public class ProjectGenerator {
...
-            let targetReference =
-                pbxProject.objects.nativeTargets.referenceValues.first { predicate($0 as PBXTarget) } ??
-                pbxProject.objects.legacyTargets.referenceValues.first { predicate($0 as PBXTarget) }!
+            let targetRerence = pbxProject.objects.targets(named: buildTarget.target).first!
 
             let buildableReference = XCScheme.BuildableReference(
                 referencedContainer: "container:\(spec.name).xcodeproj",
-                blueprintIdentifier: targetReference.reference,
-                buildableName: buildTarget.target + (targetReference.productType?.fileExtension.map { ".\($0)" } ?? ""),
+                blueprintIdentifier: targetRerence.reference,
```
Comment (rahul-malik) [inline Sources/XcodeGenKit/ProjectGenerator.swift]: Nit: Spelling of `Reference`

File: Package.swift (hunk 1 above)
Comment (yonaskolb) [reply Package.swift]: This relies on unreleased coded. It will be updated with xcproj 4.0

File: Sources/ProjectSpec/Target.swift (hunk 2 above)
Comment (yonaskolb) [reply Sources/ProjectSpec/Target.swift]: SwiftFormat did this. 
self.name is still required as name is a parameter

**Pull request merged**